File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/Model/Statements/PrintStmt.java
package Model.Statements;

import Exceptions.MyControllerException;
import Exceptions.MyStmtException;
import Model.Expressions.Exp;
import Model.Utils.MyIDictionary;
import Model.Utils.MyIHeap;
import Model.Utils.MyIList;
import Model.PrgState;

public class PrintStmt implements IStmt{
    Exp exp;
    public PrintStmt(Exp exp){this.exp=exp;}
    public String toString(){ return "print("+ exp+")";}
    public PrgState execute(PrgState state) throws MyControllerException {
        MyIList<Integer> out=state.getOut();
        MyIDictionary<String,Integer> symTbl = state.getSymTable();
        MyIHeap<Integer> heap=state.getHeap();
        try {
            out.add(exp.eval(symTbl,heap));
            return null;
        }
        catch(MyStmtException e){
            throw new MyControllerException(e);
        }
    }
}

File: 3rd Semester/Advanced Programming Methods/Laboratory/Toy Interpreter C#/sharplab/Model/Statements/PrintStmt.cs
using sharplab.Model.Expressions;
using sharplab.Model.Utils;
using System;
using System.Collections.Generic;
using System.Text;

namespace sharplab.Model.Statements
{
    class PrintStmt : IStmt
    {
        Exp exp;
        public PrintStmt(Exp exp) { this.exp = exp; }
        public override string ToString() { return "print(" + exp + ")"; }
        public PrgState Execute(PrgState state)
        {
           IMyList<int> o=state.Geto();
            IMyDictionary<string, int> symTbl = state.GetSymTable();
           try
           {
                o.Add(exp.Eval(symTbl));
                return state;
           }
           catch
           {
                throw new Exception("Invalid expression!");
           }
        }
    }
}
